cmd/juju/status/health.go

```diff
@@ -0,0 +1,252 @@
+	}
+}
+
+func (c *statusHealthCommand) SetFlags(f *gnuflag.FlagSet) {
+	f.BoolVar(&c.isoTime, "utc", false, "display time as UTC in RFC3339 format")
+}
+
+func (c *statusHealthCommand) Init(args []string) error {
+	// If use of ISO time not specified on command line,
+	// check env var.
+	if !c.isoTime {
+		var err error
+		envVarValue := os.Getenv(osenv.JujuStatusIsoTimeEnvKey)
+		if envVarValue != "" {
+			if c.isoTime, err = strconv.ParseBool(envVarValue); err != nil {
+				return errors.Annotatef(err, "invalid %s env var, expected true|false", osenv.JujuStatusIsoTimeEnvKey)
+			}
+		}
+	}
+	return nil
+}
+
+// statusHealthCommand.Run output and return code should conform to this commonly used standard
+// https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html
+func (c *statusHealthCommand) Run(ctx *cmd.Context) error {
+	var exitCode int
+	var notOkay []string
+
+	// First determine the status
+	apiclient, err := c.NewAPIClient()
+	if err != nil {
+		return fmt.Errorf(connectionError, c.ConnectionName(), err)
+	}
+	defer apiclient.Close()
+
+	status, err := apiclient.Status([]string{})
+	if err != nil {
+		// problems getting status report Unknown
+		exitCode = 3
+		notOkay = append(notOkay, err.Error())
+	}
+
+	//Parse status setting exitCode and summary message to the most severe status
+	if status != nil {
+		sf := NewStatusFormatter(status, c.isoTime)
+		formatted := sf.format()
+		code, issues := formatted.Health()
+		if code > exitCode {
+			exitCode = code
+		}
+		notOkay = append(notOkay, issues...)
+	}
+
+	//set summary line based on most sever exitcode
+	summary := "Juju Health "
+	switch exitCode {
+	case 0:
+		summary += "Okay"
+	case 1:
+		summary += "Warning"
+	case 2:
+		summary += "Critical"
+	default:
+		summary += "Unknown"
+	}
+
+	//Output the statuses which aren't okay and exit
+	fmt.Println(summary)
+	for _, line := range notOkay {
+		fmt.Println(line)
+	}
+	return cmd.NewRcPassthroughError(exitCode)
+}
+
+func (m *machineStatus) Health() (int, []string) {
+	var exitCode int
+	var notOkay []string
+	if m.Err != nil {
+		if 2 > exitCode {
+			exitCode = 2
+		}
+		notOkay = append(notOkay, m.Err.Error())
+	} else if m.AgentState != params.StatusStarted {
+		if 1 > exitCode {
+			exitCode = 1
+		}
+		notOkay = append(notOkay, fmt.Sprintf("Status: %s", m.AgentState))
+	}
+
+	for cname, cstatus := range m.Containers {
+		ccode, cissues := cstatus.Health()
+
+		if ccode > exitCode {
+			exitCode = ccode
+		}
+		for _, issue := range cissues {
+			notOkay = append(notOkay, fmt.Sprintf("Container: %s\t %s", cname, issue))
+		}
+	}
+	return exitCode, notOkay
+}
+
+func (s *serviceStatus) Health() (int, []string) {
+	var exitCode int
+	var notOkay []string
+
+	if s.Err != nil {
+		if 2 > exitCode {
+			exitCode = 2
+		}
+		notOkay = append(notOkay, s.Err.Error())
+	} else if s.StatusInfo.Current != params.StatusStarted && s.StatusInfo.Current != params.StatusIdle {
+		if 1 > exitCode {
```

Comment:
Good point I forgot about the update-status hook running on a regular basis. I was trying catch states that  I didn't specifically know about. If some work goes into clarifying the states in Juju this may no longer be needed. One way or another I'll fix it in the next update.